mempool/src/filter.rs

```diff
@@ -0,0 +1,155 @@
+use account::Account;
+use collections::LimitHashSet;
+use keys::Address;
+use nimiq_hash::{Blake2bHash, Hash};
+use primitives::coin::Coin;
+use primitives::networks::NetworkId;
+use transaction::{Transaction, TransactionFlags};
+
+
+
+#[derive(Debug)]
+pub struct MempoolFilter {
+    blacklist: LimitHashSet<Blake2bHash>,
+    rules: Rules,
+}
+
+impl MempoolFilter {
+
+    pub const DEFAULT_RULES: Rules = Rules {
+        tx_fee: Coin::ZERO,
+        tx_fee_per_byte: 0.0,
+        tx_value: Coin::ZERO,
+        tx_value_total: Coin::ZERO,
+        contract_fee: Coin::ZERO,
+        contract_fee_per_byte: 0.0,
+        contract_value: Coin::ZERO,
+        creation_fee: Coin::ZERO,
+        creation_fee_per_byte: 0.0,
+        creation_value: Coin::ZERO,
+        sender_balance: Coin::ZERO,
+        recipient_balance: Coin::ZERO,
+    };
+
+    pub const DEFAULT_BLACKLIST_SIZE: usize = 25000;
+
+    pub fn new(rules: Rules, blacklist_limit: usize) -> Self {
+        let b: LimitHashSet<Blake2bHash> = LimitHashSet::new(blacklist_limit);
+        MempoolFilter {blacklist: b, rules}
```

Comment:
Formatting
```suggestion
        MempoolFilter {
            blacklist: b,
            rules,
        }
```

Reply:
For short constructors, I usually also format them in one line. Do we have a concrete code style? I think Rust suggest a certain code style and we could use auto-format to enforce it. Anyway, maybe we should discuss this in a seperate Issue.